sugoi-api-jms-store-provider/src/main/java/fr/insee/sugoi/jms/JmsWriterStore.java

```diff
@@ -51,215 +59,315 @@ public JmsWriterStore(
     params.put(JmsAtttributes.USER_STORAGE, userStorage.getName());
-    jmsWriter.writeRequestInQueue(queueRequestName, Method.DELETE_APPLICATION, params);
+    return CheckAndSend(Method.DELETE_APPLICATION, params, applicationName, providerRequest);
   }
 
   @Override
-  public void changePassword(
-      User user, String oldPassword, String newPassword, PasswordChangeRequest pcr) {
+  public ProviderResponse changePassword(
+      String userId,
+      String oldPassword,
+      String newPassword,
+      PasswordChangeRequest pcr,
+      ProviderRequest providerRequest) {
     Map<String, Object> params = new HashMap<>();
-    params.put(JmsAtttributes.USER, user);
+    params.put(JmsAtttributes.USER_ID, userId);
     params.put(JmsAtttributes.NEW_PASSWORD, newPassword);
     params.put(JmsAtttributes.OLD_PASSWORD, oldPassword);
     params.put(JmsAtttributes.REALM, realm.getName());
     params.put(JmsAtttributes.USER_STORAGE, userStorage.getName());
     params.put(JmsAtttributes.PASSWORD_CHANGE_REQUEST, pcr);
-    jmsWriter.writeRequestInQueue(queueRequestName, Method.CHANGE_PASSWORD, params);
+    return CheckAndSend(Method.CHANGE_PASSWORD, params, userId, providerRequest);
   }
 
   @Override
-  public void addAppManagedAttribute(String userId, String attributeKey, String attribute) {
+  public ProviderResponse addAppManagedAttribute(
+      String userId, String attributeKey, String attribute, ProviderRequest providerRequest) {
     Map<String, Object> params = new HashMap<>();
     params.put(JmsAtttributes.USER_ID, userId);
     params.put(JmsAtttributes.ATTRIBUTE_KEY, attributeKey);
     params.put(JmsAtttributes.ATTRIBUTE_VALUE, attribute);
-    jmsWriter.writeRequestInQueue(queueRequestName, Method.ADD_APP_MANAGED_ATTRIBUTE, params);
+    return CheckAndSend(Method.ADD_APP_MANAGED_ATTRIBUTE, params, userId, providerRequest);
   }
 
   @Override
-  public void deleteAppManagedAttribute(String userId, String attributeKey, String attribute) {
+  public ProviderResponse deleteAppManagedAttribute(
+      String userId, String attributeKey, String attribute, ProviderRequest providerRequest) {
     Map<String, Object> params = new HashMap<>();
     params.put(JmsAtttributes.USER_ID, userId);
     params.put(JmsAtttributes.ATTRIBUTE_KEY, attributeKey);
     params.put(JmsAtttributes.ATTRIBUTE_VALUE, attribute);
     params.put(JmsAtttributes.REALM, realm.getName());
     params.put(JmsAtttributes.USER_STORAGE, userStorage.getName());
-    jmsWriter.writeRequestInQueue(queueRequestName, Method.DELETE_APP_MANAGED_ATTRIBUTE, params);
+    return CheckAndSend(Method.DELETE_APP_MANAGED_ATTRIBUTE, params, userId, providerRequest);
+  }
+
+  private ProviderResponse CheckAndSend(
+      String method, Map<String, Object> params, String entityId, ProviderRequest providerRequest) {
+    ProviderResponse response = new ProviderResponse();
+
+    // Perhaps use 2 jms template, one with a "reasonable" timeout for synchronous
```

Comment:
this is what you called queueUrgentResponseName and queueResponseName ?

Reply:
Not really, in fact we have 2 sender of broker message: one which wait for a response during x seconds and if there is no response the rest of the code is executed (synchronous), and on another one which doen't wait for a response (asynchronous).